lib/spack/spack/spec.py

```diff
@@ -4524,6 +4524,21 @@ def do_parse(self):
         except spack.parse.ParseError as e:
             raise SpecParseError(e)
 
+        # Generate lookups for git-commit-based versions
+        for spec in specs:
```

Comment:
Just thinking out loud - is it ever the case that the same package is in a list of specs (for different versions) and would there be redundant generation (or starting the run and seeing that the commits exist and just exiting). I guess that's not so bad, but I wish we could have something like:

```bash
for pkg in get_unique_packages(specs):
   ...